lib/Elastica/Aggregation/Filter.php

```diff
@@ -13,13 +14,19 @@ class Filter extends AbstractAggregation
 {
     /**
      * @param string         $name
-     * @param AbstractFilter $filter
+     * @param AbstractQuery $filter
      */
-    public function __construct($name, AbstractFilter $filter = null)
+    public function __construct($name, $filter = null)
     {
         parent::__construct($name);
 
         if ($filter !== null) {
+            if ($filter instanceof AbstractFilter) {
```

Comment:
Usually we need check filter on null if it can be null, like in this case, we set filter if it is only not null. So, we can't move it to function.

About code simplify and extract it to Param, I don't see big advantage, because function call will be with messages, and it save just 2-3 lines. And some places can have filter as array, that require another function with another checks. 

But negative effect here that exceptions and error will be triggered in Param and this will be not a last call at stack, so developer will have not a right place - where they need change something and they must check previous call stack to find real reason.

For me save 2-3 lines in temporary code will be compensated by wrong place in call stack.

If you still wish, I can do this, no problem, just confirm your wish.